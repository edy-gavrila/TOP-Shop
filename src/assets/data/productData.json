[
    {
        "id": "0",
        "title": "Bentley Bentayga",
        "imgUrl": "https://web2.wheelz.me/wp-content/uploads/2021/05/Bentley-Bentayga-Mansory-2021-9.jpg",
        "description": "This stylish car goes well with LV Handbags and Chihuahuas.",
        "price": "130,000",
        "userRating": ""
    },
    {
        "id": "1",
        "title": "BMW 7 Series",
        "imgUrl": "https://www.bmw.ca/content/dam/bmw/common/all-models/7-series/sedan/2019/inform/bmw-7series-sedan-inform-line-02-slide-02.jpg",
        "description": "A must have for every executive. Goes well with a Rolex watch.",
        "price": "87,000",
        "userRating": ""
    },
    {
        "id": "2",
        "title": "BMW X6",
        "imgUrl": "https://mediacloud.carbuyer.co.uk/image/private/s--QDVS7VFS--/v1584466393/carbuyer/car_images/bmw-x6-suv-cutout-2019.jpg",
        "description": "Are you a bad boi? Then this car is definely for you!",
        "price": "64,000",
        "userRating": ""
    },

    {
        "id": "3",
        "title": "Lamborghini Urus",
        "imgUrl": "https://www.lamborghini.com/sites/it-en/files/DAM/lamborghini/facelift_2019/model_gw/urus/09_26_graphite_lancio/family_chooser1_m.png",
        "description": "If you can afford this, Congratulations! You've won life!",
        "price": "230,000",
        "userRating": ""
    },
    {
        "id": "4",
        "title": "Mercedes-Benz GLE Coupe",
        "imgUrl": "https://www.mansory.com/sites/default/files/styles/1920x800_fullwidth_car_slider/public/2021-06/mansory-mercedes-gle-coupe-01.jpg?itok=vBiOpXxU",
        "description": "The essential accessory every suburban mum on the school run shoud have. Watch those rubber necks!",
        "price": "90,000",
        "userRating": ""
    },
    {
        "id": "5",
        "title": "Mercedes-Benz S Class",
        "imgUrl": "https://www.mansory.com/sites/default/files/styles/1920x800_fullwidth_car_slider/public/2021-05/mansory-mercedes-s-class-w223-01.jpg?itok=OgNXlJgr",
        "description": "Do you want to feel as you are driving an airplane crossed with a tank? Then this is for you!",
        "price": "95,000",
        "userRating": ""
    },
    {
        "id": "6",
        "title": "Audi Q8",
        "imgUrl": "https://imgd.aeplcdn.com/1200x900/n/cw/ec/34470/audi-q8-front-view4.jpeg?q=75",
        "description": "A perfect compromise between a middle of life crisys and a family man!",
        "price": "55,000",
        "userRating": ""
    },
    {
        "id": "7",
        "title": "Range Rover Evoque",
        "imgUrl": "https://new.img-cdn.lookers.co.uk/nvmp/CAR/94411/1ec3dc68-6ad9-4423-8d68-e4fb7a451cb7/42d04281-c500-4cdc-9f40-c5dc1d98084b?w=1200",
        "description": "A perfect gift for daddy's little prince(ss).",
        "price": "75,000",
        "userRating": ""
    },
    {
        "id": "8",
        "title": "Range Rover Velar",
        "imgUrl": "https://sx-content-labs.sixt.io/Media/8fleet-1050x600/land-rover-range-rover-velar-5d-silber-2018.png",
        "description": "This car is an important fashion statement. It oozes power, fame, money!",
        "price": "85,000",
        "userRating": ""
    },
    {
        "id": "9",
        "title": "Tesla Model X",
        "imgUrl": "https://cdn.motor1.com/images/mgl/VR69l/s3/2022-tesla-model-x.jpg",
        "description": "Nothing says more \"I have loads of dosh but I care about the environment than this beauty!\" ",
        "price": "65,000",
        "userRating": ""
    }
]