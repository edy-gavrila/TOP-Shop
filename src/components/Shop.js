import React from "react";
import productData from "../assets/data/productData.json";
import ProductCard from "./UI/ProductCard";

function Shop({ onAddItem }) {
  const products = productData.map((product) => (
    <ProductCard key={product.id} productData={product} onAddItem={onAddItem} />
  ));
  return (
    <div
      data-testid="shop-component"
      className="flex flex-wrap gap-4 items-center justify-center"
    >
      {products}
    </div>
  );
}

export default Shop;
