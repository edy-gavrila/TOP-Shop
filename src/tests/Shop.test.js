import { render, screen } from "@testing-library/react";
import { act } from "react-dom/test-utils";
import { BrowserRouter } from "react-router-dom";
import productData from "../assets/data/productData.json";
import Shop from "../components/Shop";

describe("Shop component tests", () => {
  const { container } = render(
    <BrowserRouter>
      <Shop />
    </BrowserRouter>
  );
  test("Shop component renders correctly", () => {
    expect(container).toMatchSnapshot();
  });

  test("Shop component renders same number of product cards as in the data file", () => {
    const numberOfDataItems = productData.length;
    act(async () => {
      const numberOfProductCards = await screen.findAllByTestId("product-card");
      expect(numberOfProductCards.length).toBe(numberOfDataItems);
    });
  });
});
