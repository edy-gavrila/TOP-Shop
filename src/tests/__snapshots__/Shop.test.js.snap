// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shop component tests Shop component renders correctly 1`] = `
<div>
  <div
    class="flex flex-wrap gap-4 items-center justify-center"
    data-testid="shop-component"
  >
    <div
      class=" bg-white border-slate-800 shadow-lg shadow-slate-600 p-2 md:w-[320px] h-[480px] flex flex-col justify-between"
      data-testid="product-card"
    >
      <div>
        <img
          alt="Bentley Bentayga"
          class=""
          src="https://web2.wheelz.me/wp-content/uploads/2021/05/Bentley-Bentayga-Mansory-2021-9.jpg"
        />
        <h4
          class="text-md mt-4 font-bold"
        >
          Bentley Bentayga
        </h4>
      </div>
      <p
        class="leading-none"
      >
        This stylish car goes well with LV Handbags and Chihuahuas.
      </p>
      <div
        class="text-3xl text-blue-500 font-bold"
      >
        £130,000
      </div>
      <button
        class="bg-slate-800 text-white py-2 w-full shadow-md shadow-slate-800 hover:bg-slate-900 active:scale-95 transition-all"
      >
        Add to Cart
      </button>
    </div>
    <div
      class=" bg-white border-slate-800 shadow-lg shadow-slate-600 p-2 md:w-[320px] h-[480px] flex flex-col justify-between"
      data-testid="product-card"
    >
      <div>
        <img
          alt="BMW 7 Series"
          class=""
          src="https://www.bmw.ca/content/dam/bmw/common/all-models/7-series/sedan/2019/inform/bmw-7series-sedan-inform-line-02-slide-02.jpg"
        />
        <h4
          class="text-md mt-4 font-bold"
        >
          BMW 7 Series
        </h4>
      </div>
      <p
        class="leading-none"
      >
        A must have for every executive. Goes well with a Rolex watch.
      </p>
      <div
        class="text-3xl text-blue-500 font-bold"
      >
        £87,000
      </div>
      <button
        class="bg-slate-800 text-white py-2 w-full shadow-md shadow-slate-800 hover:bg-slate-900 active:scale-95 transition-all"
      >
        Add to Cart
      </button>
    </div>
    <div
      class=" bg-white border-slate-800 shadow-lg shadow-slate-600 p-2 md:w-[320px] h-[480px] flex flex-col justify-between"
      data-testid="product-card"
    >
      <div>
        <img
          alt="BMW X6"
          class=""
          src="https://mediacloud.carbuyer.co.uk/image/private/s--QDVS7VFS--/v1584466393/carbuyer/car_images/bmw-x6-suv-cutout-2019.jpg"
        />
        <h4
          class="text-md mt-4 font-bold"
        >
          BMW X6
        </h4>
      </div>
      <p
        class="leading-none"
      >
        Are you a bad boi? Then this car is definely for you!
      </p>
      <div
        class="text-3xl text-blue-500 font-bold"
      >
        £64,000
      </div>
      <button
        class="bg-slate-800 text-white py-2 w-full shadow-md shadow-slate-800 hover:bg-slate-900 active:scale-95 transition-all"
      >
        Add to Cart
      </button>
    </div>
    <div
      class=" bg-white border-slate-800 shadow-lg shadow-slate-600 p-2 md:w-[320px] h-[480px] flex flex-col justify-between"
      data-testid="product-card"
    >
      <div>
        <img
          alt="Lamborghini Urus"
          class=""
          src="https://www.lamborghini.com/sites/it-en/files/DAM/lamborghini/facelift_2019/model_gw/urus/09_26_graphite_lancio/family_chooser1_m.png"
        />
        <h4
          class="text-md mt-4 font-bold"
        >
          Lamborghini Urus
        </h4>
      </div>
      <p
        class="leading-none"
      >
        If you can afford this, Congratulations! You've won life!
      </p>
      <div
        class="text-3xl text-blue-500 font-bold"
      >
        £230,000
      </div>
      <button
        class="bg-slate-800 text-white py-2 w-full shadow-md shadow-slate-800 hover:bg-slate-900 active:scale-95 transition-all"
      >
        Add to Cart
      </button>
    </div>
    <div
      class=" bg-white border-slate-800 shadow-lg shadow-slate-600 p-2 md:w-[320px] h-[480px] flex flex-col justify-between"
      data-testid="product-card"
    >
      <div>
        <img
          alt="Mercedes-Benz GLE Coupe"
          class=""
          src="https://www.mansory.com/sites/default/files/styles/1920x800_fullwidth_car_slider/public/2021-06/mansory-mercedes-gle-coupe-01.jpg?itok=vBiOpXxU"
        />
        <h4
          class="text-md mt-4 font-bold"
        >
          Mercedes-Benz GLE Coupe
        </h4>
      </div>
      <p
        class="leading-none"
      >
        The essential accessory every suburban mum on the school run shoud have. Watch those rubber necks!
      </p>
      <div
        class="text-3xl text-blue-500 font-bold"
      >
        £90,000
      </div>
      <button
        class="bg-slate-800 text-white py-2 w-full shadow-md shadow-slate-800 hover:bg-slate-900 active:scale-95 transition-all"
      >
        Add to Cart
      </button>
    </div>
    <div
      class=" bg-white border-slate-800 shadow-lg shadow-slate-600 p-2 md:w-[320px] h-[480px] flex flex-col justify-between"
      data-testid="product-card"
    >
      <div>
        <img
          alt="Mercedes-Benz S Class"
          class=""
          src="https://www.mansory.com/sites/default/files/styles/1920x800_fullwidth_car_slider/public/2021-05/mansory-mercedes-s-class-w223-01.jpg?itok=OgNXlJgr"
        />
        <h4
          class="text-md mt-4 font-bold"
        >
          Mercedes-Benz S Class
        </h4>
      </div>
      <p
        class="leading-none"
      >
        Do you want to feel as you are driving an airplane crossed with a tank? Then this is for you!
      </p>
      <div
        class="text-3xl text-blue-500 font-bold"
      >
        £95,000
      </div>
      <button
        class="bg-slate-800 text-white py-2 w-full shadow-md shadow-slate-800 hover:bg-slate-900 active:scale-95 transition-all"
      >
        Add to Cart
      </button>
    </div>
    <div
      class=" bg-white border-slate-800 shadow-lg shadow-slate-600 p-2 md:w-[320px] h-[480px] flex flex-col justify-between"
      data-testid="product-card"
    >
      <div>
        <img
          alt="Audi Q8"
          class=""
          src="https://imgd.aeplcdn.com/1200x900/n/cw/ec/34470/audi-q8-front-view4.jpeg?q=75"
        />
        <h4
          class="text-md mt-4 font-bold"
        >
          Audi Q8
        </h4>
      </div>
      <p
        class="leading-none"
      >
        A perfect compromise between a middle of life crisys and a family man!
      </p>
      <div
        class="text-3xl text-blue-500 font-bold"
      >
        £55,000
      </div>
      <button
        class="bg-slate-800 text-white py-2 w-full shadow-md shadow-slate-800 hover:bg-slate-900 active:scale-95 transition-all"
      >
        Add to Cart
      </button>
    </div>
    <div
      class=" bg-white border-slate-800 shadow-lg shadow-slate-600 p-2 md:w-[320px] h-[480px] flex flex-col justify-between"
      data-testid="product-card"
    >
      <div>
        <img
          alt="Range Rover Evoque"
          class=""
          src="https://new.img-cdn.lookers.co.uk/nvmp/CAR/94411/1ec3dc68-6ad9-4423-8d68-e4fb7a451cb7/42d04281-c500-4cdc-9f40-c5dc1d98084b?w=1200"
        />
        <h4
          class="text-md mt-4 font-bold"
        >
          Range Rover Evoque
        </h4>
      </div>
      <p
        class="leading-none"
      >
        A perfect gift for daddy's little prince(ss).
      </p>
      <div
        class="text-3xl text-blue-500 font-bold"
      >
        £75,000
      </div>
      <button
        class="bg-slate-800 text-white py-2 w-full shadow-md shadow-slate-800 hover:bg-slate-900 active:scale-95 transition-all"
      >
        Add to Cart
      </button>
    </div>
    <div
      class=" bg-white border-slate-800 shadow-lg shadow-slate-600 p-2 md:w-[320px] h-[480px] flex flex-col justify-between"
      data-testid="product-card"
    >
      <div>
        <img
          alt="Range Rover Velar"
          class=""
          src="https://sx-content-labs.sixt.io/Media/8fleet-1050x600/land-rover-range-rover-velar-5d-silber-2018.png"
        />
        <h4
          class="text-md mt-4 font-bold"
        >
          Range Rover Velar
        </h4>
      </div>
      <p
        class="leading-none"
      >
        This car is an important fashion statement. It oozes power, fame, money!
      </p>
      <div
        class="text-3xl text-blue-500 font-bold"
      >
        £85,000
      </div>
      <button
        class="bg-slate-800 text-white py-2 w-full shadow-md shadow-slate-800 hover:bg-slate-900 active:scale-95 transition-all"
      >
        Add to Cart
      </button>
    </div>
    <div
      class=" bg-white border-slate-800 shadow-lg shadow-slate-600 p-2 md:w-[320px] h-[480px] flex flex-col justify-between"
      data-testid="product-card"
    >
      <div>
        <img
          alt="Tesla Model X"
          class=""
          src="https://cdn.motor1.com/images/mgl/VR69l/s3/2022-tesla-model-x.jpg"
        />
        <h4
          class="text-md mt-4 font-bold"
        >
          Tesla Model X
        </h4>
      </div>
      <p
        class="leading-none"
      >
        Nothing says more "I have loads of dosh but I care about the environment than this beauty!" 
      </p>
      <div
        class="text-3xl text-blue-500 font-bold"
      >
        £65,000
      </div>
      <button
        class="bg-slate-800 text-white py-2 w-full shadow-md shadow-slate-800 hover:bg-slate-900 active:scale-95 transition-all"
      >
        Add to Cart
      </button>
    </div>
  </div>
</div>
`;
